src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/SelfcareApiServiceImpl.scala

```diff
@@ -73,4 +77,100 @@ final case class SelfcareApiServiceImpl(
+    contexts: Seq[(String, String)],
+    toEntityMarshallerUserInfo: ToEntityMarshaller[UserInfo],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(s"Retrieving user $userId")
+    val headers: List[HttpHeader] = headersFromContext()
+
+    val result: Future[UserInfo] = for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      personId       <- getUidFutureUUID(contexts)
+      userUuid       <- userId.toFutureUUID
+      selfcareUuid   <- getSelfcareIdFutureUUID(contexts)
+      usersInfoApi   <- selfcareV2ClientService
+        .getInstitutionProductUsers(
+          institutionId = selfcareUuid,
+          userId = userUuid.some,
+          requesterId = personId,
+          roles = Seq.empty
+        )
+        .map(_.map(_.toApi(organizationId)))
+
+      users    <- usersInfoApi.traverse(_.toFuture)
+      userInfo <- users.headOption.toFuture(UserNotFound(selfcareUuid, userUuid))
+    } yield userInfo
+
+    onComplete(result) {
+      case Success(userInfo)         => getUser200(headers)(userInfo)
+      case Failure(ex: UserNotFound) =>
+        logger.error(s"Error while retrieving user $userId - ${ex.getMessage}")
+        getUser404(headers)(problemOf(StatusCodes.NotFound, ex))
+      case Failure(ex)               =>
+        logger.error(s"Error while retrieving user $userId - ${ex.getMessage}")
+        complete(
+          StatusCodes.InternalServerError,
+          headers,
+          problemOf(
+            StatusCodes.InternalServerError,
+            GenericError(s"Something went wrong trying to get user $userId - ${ex.getMessage}")
+          )
+        )
+    }
+  }
+  override def getInstitutionUsers(userId: Option[String], roles: String, query: Option[String], tenantId: String)(
+    implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerUserInfoarray: ToEntityMarshaller[Seq[UserInfo]],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(s"Retrieving users for institutions $tenantId")
+    val headers: List[HttpHeader] = headersFromContext()
+
+    def filterByUserName(users: Seq[UserInfo], query: Option[String]): Seq[UserInfo] = {
+      query.fold(users)(q =>
+        users.filter(user =>
+          user.name.toLowerCase.contains(q.toLowerCase) || user.familyName.toLowerCase.contains(q.toLowerCase)
+        )
+      )
+    }
+
+    val result: Future[Seq[UserInfo]] = for {
+      userUuid <- userId.traverse(_.toFutureUUID)
+      rolesParams = parseArrayParameters(roles)
+      tenantUuid   <- tenantId.toFutureUUID
+      tenant       <- tenantProcessService.getTenant(tenantUuid)
+      selfcareId   <- tenant.selfcareId.toFuture(MissingSelfcareId(tenant.id))
+      selfcareUuid <- selfcareId.toFutureUUID
+      users        <- selfcareV2ClientService
+        .getInstitutionProductUsers(
+          institutionId = selfcareUuid,
+          userId = userUuid,
+          requesterId = tenantUuid,
```

Comment:
what should be used as `requesterId`? here I see the `tenantId`, but [here](https://github.com/pagopa/interop-be-backend-for-frontend/pull/240/files?diff=split&w=1#diff-afdc963d99c255c0f33a40498c808ff6d788e398079d39820b4d3854bc09ffa7R98) it is used the `personId`